relayer/chain/substrate/listener.go

```diff
@@ -5,30 +5,38 @@ package substrate
 
 import (
 	"context"
+	"encoding/json"
 	"time"
 
-	"golang.org/x/sync/errgroup"
-
+	"github.com/ethereum/go-ethereum/common"
 	"github.com/sirupsen/logrus"
 	rpcOffchain "github.com/snowfork/go-substrate-rpc-client/v2/rpc/offchain"
 	"github.com/snowfork/go-substrate-rpc-client/v2/types"
+	"golang.org/x/sync/errgroup"
+
 	"github.com/snowfork/polkadot-ethereum/relayer/chain"
+	"github.com/snowfork/polkadot-ethereum/relayer/store"
 	chainTypes "github.com/snowfork/polkadot-ethereum/relayer/substrate"
 )
 
 type Listener struct {
-	config   *Config
-	conn     *Connection
-	messages chan<- []chain.Message
-	log      *logrus.Entry
+	config        *Config
+	conn          *Connection
+	relayconn     *Connection
+	messages      chan<- []chain.Message
+	beefyMessages chan<- store.DatabaseCmd
+	log           *logrus.Entry
 }
 
-func NewListener(config *Config, conn *Connection, messages chan<- []chain.Message, log *logrus.Entry) *Listener {
+func NewListener(config *Config, conn *Connection, relayconn *Connection, messages chan<- []chain.Message,
```

Comment:
conn -> parachainConn

Reply:
Refactoring to separate relaychain/parachain abstractions left just one connection per abstraction, so I left this as `conn` in https://github.com/Snowfork/polkadot-ethereum/commit/b428332f96d28dd075dd6dee02c99b9f1b554f5d.